src/git-remove-all.js

```diff
@@ -3,22 +3,18 @@
 const fs = require('fs-extra');
 const path = require('path');
 const run = require('./run');
-
-async function lsFiles(options) {
-  // attempt to fix https://github.com/ember-cli/ember-cli-update/issues/583
-  options = { maxBuffer: 500 * 1024, ...options };
-
-  let files = (await run('git ls-files', options)).trim().split(/\r?\n/g);
-
-  return files;
-}
+const { runWithSpawn } = require('./run');
 
 module.exports = async function gitRemoveAll(options) {
   // this removes cwd as well, which trips up your terminal
   // when in a monorepo
   // await run('git rm -rf .', options);
 
-  let files = await lsFiles(options);
+  // Using spawn rather than run here because ls-files produces a lot
+  // of output if it gets run in a large repo
+  let files = (await runWithSpawn('git', ['ls-files'], options))
```

Comment:
Can you move this back into `lsFiles`, and mention this https://github.com/kellyselden/git-diff-apply/pull/277 for reference?

Reply:
Yes please.